code/src/components/AddTask.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import styled from "styled-components";
+
+import tasks from "../reducers/tasks";
+
+const AddTask = () => {
+  const [input, setInput] = useState("");
+
+  const dispatch = useDispatch();
+
+  const onAddTask = () => {
+    dispatch(tasks.actions.addTask(input));
+    setInput(""); // -- clears the input field when submitted
+  };
+
+  const handleKeyPress = (event) => {
+    // -- enter input with enter key
+    if (event.key === "Enter") {
+      onAddTask();
+    }
+  };
+
+  return (
+    <InputContainer>
+      <Input
+        type="text"
+        value={input}
+        onChange={(event) => setInput(event.target.value)}
+        placeholder="add new to do!"
+        onKeyPress={handleKeyPress}
+      />
+      <Button onClick={onAddTask}>
```

Comment:
I would suggest to either put required on the input field or disable this button when the user hasent written anything. Because you can push through empty todos atm!